packages/client-core/src/systems/ui/AvatarDetailView.tsx

```diff
@@ -33,10 +38,38 @@ function createAvatarDetailState(id: string) {
 }
 
 type CharacterDetailState = ReturnType<typeof createAvatarDetailState>
-
+interface TypingDetail {
+  typing: Boolean
+  userId: String
+}
 const CharacterDetailView = () => {
+  const { t } = useTranslation()
+  addActionReceptor(Engine.currentWorld.store, userTypingActionReceptor)
```

Comment:
this is an anti-pattern, we don't want to add a new receptor every time this component re-renders